### PR TITLE
Fix - AUT-2803 - Add mocks for ExtensionManager to pass the tests

### PR DESCRIPTION
**Pull request by Karol-Stelmaczonek**
## Changelog
Add mocks for ExtensionManager and SM to pass the tests
Remove deprecated code usage
Move test xmls from code to files
## How to test
Run phpunit with the test

### CODE REVIEW

File: test/unit/model/qti/ItemTest.php [hunk 1]
```diff
@@ -21,89 +21,80 @@
...
-        if (!defined('EXTENSION_PATH')) {
-            define('EXTENSION_PATH', ROOT_URL);
-        }
-
-        ServiceManager::getServiceManager()->overload(
-            ApplicationService::SERVICE_ID,
-            $this->createMock(ApplicationService::class)
-        );
-    }
+        define('PRODUCT_NAME', self::PRODUCT_NAME);
```
Comment (mike-ionut-mihai-sandu-tao) [inline test/unit/model/qti/ItemTest.php:42]: please don't create constant in this way

File: test/unit/model/qti/ItemTest.php [hunk 2]
```diff
@@ -21,89 +21,80 @@
...
-
-    
-    
-    </assessmentItem>
-
-ITEM_QTI;
-
+        $expectedItemQti = file_get_contents(
+            dirname(
+                __DIR__,
```
Comment (mike-ionut-mihai-sandu-tao) [inline test/unit/model/qti/ItemTest.php:89]: please don't use native php functions ,  rather use our abstractions

**Review by mike-ionut-mihai-sandu-tao — CHANGES_REQUESTED** [review]
please take a another look at the standards and common agreements inside AUT team

File: test/unit/model/qti/ItemTest.php (hunk 1 above)
Comment (Karol-Stelmaczonek) [reply test/unit/model/qti/ItemTest.php:42]: I tried to found how to declare it or import it but can see it in other unit tests. Old tests use just define as I did. Can you give me some exaple?

File: test/unit/model/qti/ItemTest.php (hunk 2 above)
Comment (Karol-Stelmaczonek) [reply test/unit/model/qti/ItemTest.php:89]: Can you suggest me something to load file? Other tests use Documetnt class or other xml parsers but I just want to load file contents to use it in assertion.

File: test/unit/model/qti/ItemTest.php (hunk 1 above)
Comment (pnal) [reply test/unit/model/qti/ItemTest.php:42]: I guess we still define all this constants except `PRODUCT_NAME` in generis bootstrap or where ever so we can move it there as well

File: test/unit/samples/model/qti/item/testToQTIWithDirAttributeInItemBody_expectedItemQti.xml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?><assessmentItem
+        xsi:schemaLocation=""
+     identifier="Item_1" title="" label="" adaptive="false" timeDependent="false" toolName="TAO">
+
+    
+
```
Comment (pnal) [inline test/unit/samples/model/qti/item/testToQTIWithDirAttributeInItemBody_expectedItemQti.xml:6]: Do we need these empty lines?

File: test/unit/model/qti/ItemTest.php (hunk 2 above)
Comment (pnal) [reply test/unit/model/qti/ItemTest.php:89]: I think it is quite acceptable to use native function for reading some sample data.

**Review by bartlomiejmarszal — CHANGES_REQUESTED** [review]
**Review Checklist**
- [x] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful
- [ ] Pull request's target is not `master`
- [ ] Commits are following [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [ ] Commits messages are meaningful
- [ ] Commits are atomic
- [ ] Changelog is updated according to changes (if applicable)
- [ ] Documentation is updated according to changes (if applicable)

File: test/unit/model/qti/ItemTest.php (hunk 1 above)
Comment (Karol-Stelmaczonek) [reply test/unit/model/qti/ItemTest.php:42]: I talk with @mike-ionut-mihai-sandu-tao and I just need to wrap it with IF

File: test/unit/model/qti/ItemTest.php (hunk 2 above)
Comment (Karol-Stelmaczonek) [reply test/unit/model/qti/ItemTest.php:89]: I already have version using flysystem so I can commit it or scrap it and leave native functions IDK.

File: test/unit/samples/model/qti/item/testToQTIWithDirAttributeInItemBody_expectedItemQti.xml (hunk 3 above)
Comment (Karol-Stelmaczonek) [reply test/unit/samples/model/qti/item/testToQTIWithDirAttributeInItemBody_expectedItemQti.xml:6]: Method whitch we are testing toQti() returns xml is such format with empty lines and so if we want to compare it we need those lines in this saimple file.

File: test/unit/model/qti/ItemTest.php (hunk 1 above)
Comment (Karol-Stelmaczonek) [reply test/unit/model/qti/ItemTest.php:42]: > We could move it to a generis bootstrap file.

We decided together not to do that and I will stick to it.

**Review by pnal — CHANGES_REQUESTED** [review]
todo: please change target branch
todo: please follow conventional commits

File: test/unit/samples/model/qti/item/testToQti_expectedItemQti.xml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?><assessmentItem
+        xsi:schemaLocation=""
+     identifier="Item_1" title="" label="" adaptive="false" timeDependent="false" toolName="TAO">
+
+    
+    
+
```
[the remark this answers is not in the document]
Comment (Karol-Stelmaczonek) [reply test/unit/samples/model/qti/item/testToQti_expectedItemQti.xml:7]: Nikita alredy asked about it https://github.com/oat-sa/extension-tao-itemqti/pull/2294#discussion_r1045513089

**Review by pnal — APPROVED** [review]

**Pull request merged**